design/ceph/ceph-cluster-cleanup.md

```diff
@@ -0,0 +1,65 @@
+# Ceph cluster clean up policy
+
+## Use case
+
+As a rook user, I want to clean up the datadirhostpath after I intentionally uninstall ceph cluster , so that I can start a new cluster without having to do any manual clean up.
+
+## Background
+
+If the user deletes a cluster and starts a new cluster on the same hosts, the path used by dataDirHostPath must be deleted. Otherwise, stale keys and other config will remain from the previous cluster and the new mons will fail to start. As of now, the user has to manually delete the dataDirHostPath on each host. This implementation aims to automate this deletion.
+
+## Design Flow
+
+### Preconditions
+
+- Operator receives a delete request when the user deletes the ceph cluster.
+- **Important**: User confirmation is required before cleaning up the dataDirHostPath. This is important because user might have accidentally deleted the CR and in that case cleaning up the hostpath won’t recover the cluster. This final user confirmation can be a flag/label on the cephCluster
+  Adding the confirmation labels will cause the operator to refuse running an orchestration.
+- Cluster clean up should start when DeletionTimeStamp is observed on the cephCluster CR
+- Cleanup should start once the Finalizers are executed (removed).
+
+### Cleanup Process
+
+- Identify the nodes where ceph daemon are running. This should be done before the finalizers are executed.
+- Create a batch job that runs on each of the above nodes.
+- The job execution should wait till all the ceph daemons are destroyed. This is important because deleting hostpath before the daemons would cause the daemons to panic.
+- Once the ceph daemons are deleted on the node, the job should be executed to perform:
+  - dataDirHostPath clean up on all the nodes.
+
+## Cleanup Job Spec:
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: rook-ceph-cleanup-<node-name>
+spec:
+  template:
+    spec:
+      containers:
+        - name: rook-ceph-cleanup-<node-name>
+          securityContext:
+            privileged: true
+          image: <rook-image>
+          env:
+          # if ROOK_DATA_DIR_HOST_PATH is available, then delete the dataDirHostPath
+          - name: ROOK_DATA_DIR_HOST_PATH
+            value: <dataDirHostPath>
+          args: []string{"ceph", "clean"}
+          volumeMounts:
+            - name: cleanup-volume
+              # data dir host path that needs to be cleaned up.
+              mountPath: <dataDirHostPath>
```

Comment:
How about the following approach?
- This volumeMount is only added if there is a host path to delete
- The `mountPath` can be a hard-coded path such as `/rook-path-to-delete`. (This path can be different from the actual path on the host defined with the `volume` below.)
- If that path is found inside the job, it knows to clean it. If it is not found, it was not mounted, so no need to clean it.